Preliminary: FEDOT added

nicl-nno opened the pull request.

nicl-nno committed:
FEDOT added

IIaKyJIuH commented on frameworks/FEDOT/exec.py:
```diff
@@ -0,0 +1,79 @@
+
+def save_artifacts(chain, config):
+    try:
+        artifacts = config.framework_params.get('_save_artifacts', False)
+        if 'models' in artifacts:
```
Тапл пустой в качестве дефолтного параметра нужен, иначе будет ошибка в случае `False`.

nicl-nno replied:
Поправил

IIaKyJIuH commented on frameworks/FEDOT/exec.py:
```diff
@@ -0,0 +1,79 @@
+        r2='r2',
+        rmse='rmse'
+    )
+    scoring_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
```
Короче будет так: `metrics_mapping.get(config.metric, None)`

nicl-nno replied:
Поменял

IIaKyJIuH commented on frameworks/FEDOT/exec.py:
```diff
@@ -0,0 +1,79 @@
+    )
+    scoring_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+
+    if scoring_metric is None:
+        raise ValueError(f'Performance metric {config.metric} not supported.')
```
Так может геттер для `scoring_metric` поместить просто в try...catch, чтобы там же как раз и пробросить ошибку?

nicl-nno replied:
Заменил на вывод лог по аналогии с другими фреймворками

IIaKyJIuH commented on frameworks/FEDOT/exec.py:
```diff
@@ -0,0 +1,79 @@
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
```
Тут тоже лучше типизировать

nicl-nno replied:
В остальных фреймворках не типизировано, лучше сделаю единообразно.

IIaKyJIuH commented on frameworks/FEDOT/__init__.py:
```diff
@@ -0,0 +1,39 @@
+
+
+def run(dataset: Dataset, config: TaskConfig):
+    from frameworks.shared.caller import run_in_venv
```
Почему этот импорт тут?

nicl-nno replied:
Какая-то специфика реализации запуска в отдельном процессе.

IIaKyJIuH commented on frameworks/FEDOT/__init__.py:
```diff
@@ -0,0 +1,39 @@
+        )
+    )
+
+    def process_results(results):
```
Нужна типизация, тогда, кажется, коммент ниже убирается.

nicl-nno replied:
Тут тоже аналогия с имеющимися.

IIaKyJIuH commented on frameworks/FEDOT/exec.py:
```diff
@@ -0,0 +1,79 @@
+
+    is_classification = config.type == 'classification'
+    # Mapping of benchmark metrics to FEDOT metrics
+    metrics_mapping = dict(
+        acc='acc',
+        auc='roc_auc',
+        f1='f1',
+        logloss='logloss',
+        mae='mae',
+        mse='mse',
+        msle='msle',
+        r2='r2',
+        rmse='rmse'
+    )
```
Много лишнего ради одной несостыковки, короче никак?

nicl-nno replied:
Думаю для большей наглядности примера лучше полную таблицу соответствия иметь

PGijsbers committed:
Upgrade Python version and dependencies (#520)

* Remove usage of np.float alias and just use float

* Bump to Py3.9

* Update requirements for March 2023, Py3.9

* Pin packaging, since LegacyVersion was removed.

Also remove scipy pin, since later autosklearn needs higher scipy.

* Install packages to ranger/lib

* Set secret PAT used when installing with R remotes

Specifically for mlr3automl integration

* Update usage for oct 21 release

* Disable custom installed packages

* Remove installation of reqiurements altogether

* Insert oboe example

* Add monkeypatch

* Make error matrix numpy array

* Upgrade to Ubuntu 22.04 from 18.04

* Update pip cache to look at 3.9 directory

* Add Github PAT to run_all_frameworks script

* bump github action versions

TrellixVulnTeam committed:
Adding tarfile member sanitization to extractall() (#508)

eddiebergman committed:
Included lightautoml in frameworks_stable (#412)

* Included lightautoml in frameworks_stable

* Added MLNet to frameworks_latest

* Added mlr3 to both stable and latest

* copy/paste fix

PGijsbers committed:
Remove travis file (#529)

* Remove travis file since it is not used

* Update readme to reflect Python 3.9 support

* Add github action workflow to replace old travis file

* Add job id, improve name

* Fix bug where task inference would lead to KeyError

* Update type data for new openml/pandas

Probably ought to remove the specific check if we don't enforce it.

* Write numeric categories as str, see renatopp/liac-arff/issues/126

PGijsbers committed:
[Open for review] Store results after each job completion (#526)

* ensure that results are solved progressively in all situations instead of only when all jobs are completed

* rename config flag

* don't forget to cleanup job runner exec thread

* Improve type hints

* Adding file lock on global results file (#453)

* adding file lock on global results file

* fix imports

* fix amlb.utils export

* cosmetic

* clranup util imports (also magic strings) + remove ruamel dependency in subprocesses

---------

Co-authored-by: Sebastien Poirier <sebastien@h2o.ai>

* Update the requirements files to exclude yaml and include filelock

The remainder of dependencies are not re-generated to avoid
additional changes in the PR.

* Add missing import

* Add fallback for when job is not started

* Return an empty dataframe if dataframe is empty

This avoids a bug where an empty dataframe is indexed.

* Inform the user result summary is not available in AWS mode

As results are processed in a different manner (files are directly
copied over from S3). This avoids a bug where a benchmark
results.csv file tries to be accessed.

* Separate scoreboard generation to two lines instead

Which makes it easier to tell which part of the generation generates
an error, if any.

* re-enable logging

* Provide a warning and return early if no process output is detected

This avoids potentially crashing if the logging is configured incorrectly.
In the future, we should expand this to first check how logging is
configured in order to see whether or not the issue should be reported
and possibly give a more detailed warning if it is likely the cause
of an error.

---------

Co-authored-by: Sebastien Poirier <sebastien@h2o.ai>
Co-authored-by: seb <sebastien.poirier@h2o.ai>

PGijsbers committed:
maint: upgrade AMI to Ubuntu 22.04 #512 (#525)

PGijsbers committed:
Add `flaml_benchmark` (#528)

* dont discard setup_args if it already is a list

* Add flaml and flaml_benchmark

It is not added to latest since install from latest seems to be broken

* Set up alternative way for benchmark mode of flaml

This is only temporarily allowed - we expect an easily configurable
algorithm, instead of having to carefully install specific
dependencies.

* limit install, since >2 incompatible

PGijsbers committed:
Measure inference time (#532)

Add the option to measure inference time (disabled by default) for most frameworks.
For those frameworks, inference time is measured capturing both the data loading and the inference.
This is done to make things more equal between the different frameworks (as some _need_ to read the file if they don't operator in Python). Inference time is measured multiple times for different batch sizes (configurable). By default, the median is reported in the results file (as it is less sensitive to e.g., cold-starts) but all measured inference times are stored in the predictions folder of a run.
For Python frameworks, inference time for in-memory single row predictions is also measured.

PGijsbers committed:
Upload to OpenML (#523)

Adds a script that allows uploading run results to openml.
Additional metadata is stored in the task information to be able to provide a complete description for openml upload.
Additional parameters are added to `run_benchmark` to allow runs to automatically be tagged, and to connect to the test server.
Also fixes TPOT integration for newer versions, where if a model has no `predict_proba` an `AttributeError` is raised instead of a `RuntimeError`.

PGijsbers committed:
Fix a race condition of checking vs adding results (#535)

Specifically, adding results was queued in a job executor, while
checking results was directly called by the worker threads.
If the worker thread checks before the executor had added results,
it is possible to get into a deadlock condition. The deadlock
arises from the fact that the `stop` condition is never called
and the main thread will continue to wait for its END_Q signal.

PGijsbers committed:
Add scikit_safe inference time measurement files (#537)

* Add scikit_safe inference time measurement files

These files have categorical values numerically encoded and missing
values imputed, which makes them usable for any scikit-learn algo.

* Only generate inference measurement files if enabled

PGijsbers committed:
Optionally limit inference time measurements by dataset size (#538)

PGijsbers committed:
Add versions 2023 q2 (#539)

* Fix versions for June 2023 benchmark

* Add 2023Q2 framework tag

* Use encoded values for inference

* Add us-east-2 AMI

* Run docker as root on AWS

* Add option to add build options for docker build command

* Remove 'infer_speed' artifact as it is not supported in main repo

* Fix pandas 2 not compatible with autosklearn 2 see askl#1672

PGijsbers committed:
Fix: point to the right version file (#541)

Innixma committed:
Update AutoGluon `max_memory` from 0.1 to 0.4 in persist_models (#543)

Innixma committed:
Add `optimize_for_deployment` for AutoGluon_hq (#544)

PGijsbers committed:
Reduce training time by 10% if a high_quality preset is used (#546)

* Reduce training time by 10% if a high_quality preset is used

High quality presets perform a post-fit step which takes 10~15%
of total time (by Nick's estimate). To ensure comparisons stay
reasonably fair we pre-emptively tell AutoGluon to use less time,
so that all frameworks' models are based on "max_total_time"
amount of effort.

* Allow preset to be str or list and still reduce if hq or gq

PGijsbers committed:
Add identical markers to identify fit/inferencetime/predict stages (#548)

PGijsbers committed:
Add start_time, stop_time and log_time to failure.csv (#547)

This helps more quickly identify at what stage the failure took place. E.g., if it's just a few minutes in, it is probably setup failure (such as connectivity issues).

PGijsbers committed:
Docker/permissions (#550)

* Remove ownership changing and starting as user for docker images

Since the USER is overwritten by `-u` for non-Windows platforms,
which creates issues when the account running the docker image
is not the same as the one that created it.

* Dont run docker as root since images no longer have associated user

* Ignore some additional files not needed to run the benchmark

* Create root dir if it does not exist

This is required, because otherwise in docker mode a non-existent
directory is mounted, which is by default locked to `root`
permissions. This in turn makes the benchmark app unable to create
the subdirectories when the image is run as user.

* Further remove user info from docker build and add run_as option

The run_as option is then configurable so that it can be enabled
for people who run into issues. Unfortunately, I observed
different behavior from two systems with the same OS and docker
versions installed. So for now I give up on one unified solution.

PGijsbers committed:
Update GAMA for v23.0.0 (#551)

PGijsbers committed:
Sync stable-v2 back into master (#554)

PGijsbers committed:
Fix/inference with empty columns (#555)

* Correct for columns with all missing values when generating batches

* Add option to keep empty features around

* Add logging markers to TunedRandomForest

PGijsbers committed:
Only list bool as numerical if OpenML has it listed as numerical (#556)

PGijsbers committed:
Update 2023 q2 (#558)

* New model_repr is a dictionary which is not compatible with f.write

* Add VolumeLimitExceeded to retry lists

* Save additional information by default

PGijsbers committed:
Add Naive AutoML integration (#549)

* Add Naive AutoML integration

* Add naiveautoml

* Add naiveautoml

* Set reasonable defaults for greater time budgets

NAML isnt designed to run for a long time, so the defaults of 20
iterations or 10 second execution timeout is not useful when
timeout is an hour. Based on a conversation with @fmohr we set
more reasonable defaults, leading to behavior which mostly
should make use of the runtime constraints.
I added an easy option to revert to normal defaults by
setting `_use_default_time_and_iterations` as a configuration
option, which will ignore any time constraints or dynamic
defaults derived from it.

* Fix: correctly log when time limits are set to default

* Allow installation from commit hash

* Use version with bugfix for longer than usual runs

nicl-nno committed:
Code review changes

nicl-nno committed:
Empty line added to setup.sh

nicl-nno committed:
Code reivew fixes

shchur committed:
Time series forecasting improvements: metrics, dataset preprocessing & AutoGluon wrapper (#564)

* Update time series benchmarking

* Update install script

* Update TS benchmarking

* Update TS

* Make seasonality required argument

* Fix quantile_levels and wql loss

* Fix point forecast used by AG

* Address PR comments

* Move time series logic to TimeSeriesDataset

* Update HOWTO.md

* Fix import order

* Address PR comments

* Add test for time series

* Add tests for time series

* Update resources/benchmarks/timeseries.yaml

PGijsbers committed:
Add Documentation Pages with MkDocs-Material (#565)

This makes documentation more readable and easier to navigate.

PGijsbers committed:
Start adding static type checking (#567)

* Remove commented out code

* Specify mypy python version

* Address mypy errors

* Fix mypy errors

* Fix mypy errors

* Add type annotations

* Add type checking requirements

* Add ignore for [attr-defined], raised by normal Namespace use

PGijsbers committed:
Add back a readme file (#577)

PGijsbers committed:
Make openml 0.14.0 and up compatible (#579)

Innixma committed:
Update README.md (#572)

* Update README.md
* Fix AGTS typo + fix item_id bug (#11)


Co-authored-by: Nick Erickson <innixma@gmail.com>
Co-authored-by: Oleksandr Shchur <oleks.shchur@gmail.com>

nicl-nno committed:
Code review fix 2
